src/groupeddataframe/groupeddataframe.jl

```diff
@@ -296,8 +301,29 @@ function Base.NamedTuple(key::GroupKey)
     N = NamedTuple{Tuple(groupcols(parent(key)))}
     N(_groupvalues(parent(key), getfield(key, :idx)))
 end
-Base.Tuple(key::GroupKey) = values(key)
 
+"""
+    copy(key::GroupKey)
+
+Construct a `NamedTuple` with the same contents as the [`GroupKey`](@ref).
+"""
+Base.copy(key::GroupKey) = NamedTuple(key)
+
+Base.convert(::Type{NamedTuple}, key::GroupKey) = NamedTuple(key)
+Base.convert(::Type{Tuple}, key::GroupKey) = Tuple(key)
+
+Base.convert(::Type{Vector}, key::GroupKey) = [v for v in key]
```

Comment:
FWIW `NamedTuple` doesn't implement these methods. Maybe they should be defined for any iterator in Base. But I guess it's OK to provide them here.

Reply:
These are implemented, because `DataFrameRow` implements them.